src/components/avatar/index.js

```diff
@@ -142,28 +156,48 @@ export default class Avatar extends BaseComponent {
 
   getStatusBadgeColor(status) {
     switch (status) {
-      case Avatar.modes.NONE:
-        return null;
+
       case Avatar.modes.AWAY:
         return Colors.yellow30;
       case Avatar.modes.ONLINE:
         return Colors.green30;
       case Avatar.modes.OFFLINE:
         return Colors.dark60;
+      case Avatar.modes.NONE:
       default:
         return null;
     }
   }
 
-  getBadgeColor(isOnline, status) {
-    const onlineOverride = status === STATUS_MODES.NONE ? isOnline : false;
-    const badgeColor = onlineOverride ? Colors.green30 : this.getStatusBadgeColor(status);
-    return badgeColor;
+  getBadgeColor() {
+    const {isOnline, status} = this.props;
+    const statusColor = this.getStatusBadgeColor(status);
+    const onlineColor = isOnline ? Colors.green30 : undefined;
+
+    return _.get(this.props, 'badgeProps.backgroundColor') || statusColor || onlineColor;
   }
-  
+
   getBadgeSize = () => _.get(this.props, 'badgeProps.size', DEFAULT_BADGE_SIZE);
-  
-  getBadgeBorderWidth = () => _.get(this.props, 'badgeProps.borderWidth', DEFAULT_BADGE_BORDER_WIDTH);
+
+  getBadgeBorderWidth() {
+    const {badgeProps} = this.props;
+    const borderWidths = _.get(this.props, 'badgeProps.borderWidth', DEFAULT_BADGE_BORDER_WIDTH);
+
+    if (_.get(this.props, 'badgeProps.icon')) {
+      return badgeProps.borderWidth || 0;
+    }
+    return borderWidths;
+  }
+
+  getBorderColor() {
+    const {badgeProps} = this.props;
+    const borderColors = _.get(this.props, 'badgeProps.borderColor', DEFAULT_BADGE_BORDER_COLOR);
```

Comment:
again, plural. 
also, not sure I understand the logic of both `getBadgeBorderWidth` and `getBorderColor` 
why not just pass the badgeProps as is to the `Badge` component.

Reply:
It's meant to give icon badge a specific treatment but in the case of border color since in `getBorderWidth()` icon badge gets 0 if undefined I don't see a harm in passing  DEFAULT_BADGE_BORDER_COLOR (which is white) since it won't show anyway.